### ISSUE TITLE
IE Times out when trying get cookies

### ISSUE THREAD

**Issue by MemnochDaCoder** [SeleniumHQ/selenium#2547]
## Meta -

OS:  Windows 10 Pro 64-bit

Selenium Version:  
2.53.1

Browser:  
Internet Explorer

Browser Version:  
11.0.10240.16384 64-bit
## Expected Behavior -

Should get the cookies from the site.

Of note, IE is setup correctly and works for pretty much everything else I've tried to do.
## Actual Behavior -

When trying to retrieve cookies gets a timeout error.

OpenQA.Selenium.WebDriverException was unhandled by user code
  HResult=-2146233088
  Message=The HTTP request to the remote WebDriver server for URL http://localhost:16639/session/be71f8d4-b65c-44e9-af83-f13b3617742b/cookie timed out after 60 seconds.
  Source=WebDriver
  StackTrace:
       at OpenQA.Selenium.Remote.HttpCommandExecutor.CreateResponse(WebRequest request)
       at OpenQA.Selenium.Remote.HttpCommandExecutor.Execute(Command commandToExecute)
       at OpenQA.Selenium.Remote.DriverServiceCommandExecutor.Execute(Command commandToExecute)
       at OpenQA.Selenium.Remote.RemoteWebDriver.Execute(String driverCommandToExecute, Dictionary`2 parameters)
       at OpenQA.Selenium.Remote.RemoteCookieJar.GetAllCookies()
       at OpenQA.Selenium.Remote.RemoteCookieJar.get_AllCookies()
       at WebAutomation.Tests.SeleniumWebAutomationTests.TestIeClickNewTabCookies() in C:\Users\Menoh\OneDrive\Documents\Visual Studio 2015\Projects\NetLabs\WebAutomationTests1\SeleniumWebAutomationTests.cs:line 964
  InnerException: 
       HResult=-2146233079
       Message=The request was aborted: The operation has timed out.
       Source=System
       StackTrace:
            at System.Net.HttpWebRequest.GetResponse()
            at OpenQA.Selenium.Remote.HttpCommandExecutor.CreateResponse(WebRequest request)
       InnerException: 
## Steps to reproduce -

```
[TestMethod]
    public void TestIeClickNewTabCookies()
    {
        const string navURL = @"http://www.aidata.me/";
        const string xPath = @"/html/body/div[1]/div[3]/div/div/div[2]/div/div/div[1]/a[3]";

        var driver = new InternetExplorerDriver();
        driver.Navigate().GoToUrl(navURL);

        var parentHandle = driver.CurrentWindowHandle;

        var cookies = driver.Manage().Cookies.AllCookies.ToList(); //<= Error here

        var myCookies =
            driver.SwitchTo()
                .Window(childHandle)
                .Manage()
                .Cookies.AllCookies.ToList();

        driver.Close();
        driver.Quit();

        Assert.IsNotNull(myCookies);
    }    
```

**Comment by MemnochDaCoder** [comment]
The BuildFromExtensionCookie method can be removed, just realized it was there after re-reading.

**Comment by jimevans** [comment]
What happens when you use the 32-bit IE driver?

**Comment by zenonwch** [comment]
Result for x32 posted here: https://github.com/SeleniumHQ/selenium/issues/1834

**Comment by MemnochDaCoder** [comment]
That IS with the 32-bit driver.  It's now throwing an error
System.InvalidOperationException: Could not retrieve cookies. The most common cause of this error is a mismatch in the bitness between the driver and browser. In particular, be sure you are not attempting to use a 64-bit IEDriverServer.exe against IE 10 or 11, even on 64-bit Windows.
I have triple checked to make sure that I'm using the 32-bit driver version.

**Comment by zenonwch** [comment]
I'm using IE x32, IEDriverServer_x32 on Windows 7 x64.

**Comment by MemnochDaCoder** [comment]
Trace from driver 
[IEDriverLog.txt](https://github.com/SeleniumHQ/selenium/files/544550/IEDriverLog.txt)

**Comment by MemnochDaCoder** [comment]
This is on a Windows 10 64-bit machine and the configuration from the driver page has been followed on each step including unchecking the Enable Enhanced mode under security.  The browser that I see launching seems to be the 32-bit version not the 64-bit version that is being reported in the log.

**Comment by MemnochDaCoder** [comment]
I've dug into the log here to find out what's going on.  The driver I'm using the is the 32 bit version so it should be launching the internet explorer located in the x86 folder (32 bit).  This doesn't seem to be the case since during execution checking to see which version is launching in Task Manager it seems to be launching the 64 bit version thus getting the bitness error.  Is there a work around, update, different way to launch, etc.?

**Comment by MemnochDaCoder** [comment]
I've also tried this on a server 2012 R2 machine and the same issue occurs.

**Comment by jimevans** [comment]
It's probable this issue is fixed by revision 318257c. Given the age of this issue, I'm going to close it. @MemnochDaCoder, I'm assuming you've moved on from this particular issue, but if you'd like to see if it's fixed, you're welcome to do so. The next widely available version of the IE driver (3.15) will have this fix.
